yarn-project/types/src/interfaces/state_provider.ts

```diff
@@ -0,0 +1,47 @@
+import { CONTRACT_TREE_HEIGHT, Fr, L1_TO_L2_MSG_TREE_HEIGHT, PRIVATE_DATA_TREE_HEIGHT } from '@aztec/circuits.js';
+
+import { L1ToL2MessageAndIndex } from '../l1_to_l2_message.js';
+import { MerkleTreeId } from '../merkle_tree_id.js';
+import { SiblingPath } from '../sibling_path.js';
+
+/**
+ * Interface providing methods for retrieving information about content of the state trees.
+ */
+export interface StateInfoProvider {
```

Comment:
Could this be shortened? Just `StateProvider`

Reply:
That was my initial idea. But it is not really providing any state, but information about the state, e.g., not providing the leafs, but information related to the leaf such as the index and its path.